src/Workspaces/Remote/ServiceHub/ProcessExtensions.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    /// <summary>
+    /// Boost performance of any servicehub service which is invoked by user explicit actions
+    /// </summary>
+    internal static class ProcessExtensions
+    {
+        private static bool s_settingPrioritySupported = true;
+
+        public static bool TrySetPriorityClass(this Process process, ProcessPriorityClass priorityClass)
+        {
+            if (!s_settingPrioritySupported)
+            {
+                return false;
+            }
+
+            try
+            {
+                process.PriorityClass = priorityClass;
+                return true;
+            }
+            catch (Exception e) when (e is PlatformNotSupportedException || e is Win32Exception)
```

Comment:
Does `Win32Exception` mean we should never try again? The docs aren't really clear, but it seems like maybe that could be a temporary failure.

Reply:
I wasn't sure of all the possible error codes of the underlying implementation, which is why I initially had no latch to disable future calls.

On macOS, Win32Exception is mostly sent due to invalid pid or lack of permissions, which won't change over time.